### PR TITLE
SDD-871 create course delete upload page

### PR DESCRIPTION
**Pull request by MrKevJoy**
- Create delete course upload page.
- Delete success page.
- Tests

### CODE REVIEW

File: src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs [hunk 1]
```diff
@@ -70,6 +70,40 @@ Task DeleteBlob()
...
+            using (var dispatcher = _sqlQueryDispatcherFactory.CreateDispatcher())
+            {
+                var courseUpload = await dispatcher.ExecuteQuery(new GetLatestUnpublishedCourseUploadForProvider()
+                {
+                    ProviderId = providerId
+                });
+
+                if (courseUpload == null)
+                {
+                    throw new InvalidStateException(InvalidStateReason.NoUnpublishedVenueUpload);
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs:84]:
```suggestion
                    throw new InvalidStateException(InvalidStateReason.NoUnpublishedCourseUpload);
```

File: src/Dfc.CourseDirectory.Core/DataManagement/IFileUploadProcessor.cs [hunk 2]
```diff
@@ -14,6 +14,7 @@ public interface IFileUploadProcessor
         // Courses
         Task<SaveFileResult> SaveCourseFile(Guid providerId, Stream stream, UserInfo uploadedBy);
         Task ProcessCourseFile(Guid courseUploadId, Stream stream);
+        Task DeleteCourseUploadForProvider(Guid ProviderId);
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.Core/DataManagement/IFileUploadProcessor.cs:17]:
```suggestion
        Task DeleteCourseUploadForProvider(Guid providerId);
```

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/DeleteUpload.cshtml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+﻿@model Dfc.CourseDirectory.WebV2.Features.DataManagement.Courses.DeleteUpload.Command
+@{
+    ViewContext.SetLayoutData(title: "Data Upload - Cancel course data upload");
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/DeleteUpload.cshtml:3]: `<title>` usually matches the `<h1>`
```suggestion
    ViewContext.SetLayoutData(title: "Delete course data upload");
```

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/DeleteUploadSuccess.cshtml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+﻿@{
+    ViewBag.Title = "Data Upload - Course data upload cancelled";
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/DeleteUploadSuccess.cshtml:2]:
```suggestion
    ViewBag.Title = "Course data upload deleted";
```

Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/DeleteUploadSuccess.cshtml]: done

File: src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/DeleteUpload.cshtml (hunk 3 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.WebV2/Features/DataManagement/Courses/DeleteUpload.cshtml]: done

File: src/Dfc.CourseDirectory.Core/DataManagement/IFileUploadProcessor.cs (hunk 2 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/IFileUploadProcessor.cs]: done

File: src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs (hunk 1 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.Core/DataManagement/FileUploadProcessor.Courses.cs]: done

**Review by gunndabad — APPROVED** [review]

**Pull request merged**